number_theory/modular_unittest.cpp

```diff
@@ -73,5 +74,82 @@ TEST(ModularTest, Basic) {
   test_modular_basic<uint64_t>();
 }
 
+template <typename T>
+void test_modular_operators() {
+  using Mod10 = Modular<T(10)>;
+
+  // test addition
+  Mod10 a = 7, b = 5;
+  EXPECT_EQ(a + b, 2);
+  EXPECT_EQ(a + 3, 0);
+  EXPECT_EQ(3 + a, 0);
+  b += a;
+  EXPECT_EQ(b, 2);
+  a += 4;
+  EXPECT_EQ(a, 1);
+
+  // test subtraction
+  a = 7, b = 5;
+  EXPECT_EQ(b - a, 8);
+  EXPECT_EQ(a - 8, 9);
+  EXPECT_EQ(0 - a, 3);
+  b -= a;
+  EXPECT_EQ(b, 8);
+  a -= 10;
+  EXPECT_EQ(a, 7);
+
+  // test multiplication
+  a = 7, b = 5;
+  EXPECT_EQ(a * b, 5);
+  EXPECT_EQ(a * 2, 4);
+  EXPECT_EQ(3 * a, 1);
+  b *= a;
+  EXPECT_EQ(b, 5);
+  a *= 10;
+  EXPECT_EQ(a, 0);
+
+  // test comparison
+  EXPECT_EQ(Mod10(1), 11);
```

Comment:
This made me reconsider implicit conversion. `Mod10(1)` being equal to `int(1)` makes sense. However, `Mod10(1)` being equal to `11` makes things a little bit confusing. What's your opinion about it? Should we be concerned about this kind of usage?

Reply:
I agree. We can ask the operator== to compare the raw values, but will it make our operators a bit inconsistent?